gateway/ln-gateway/src/gatewayd/actor.rs

```diff
@@ -57,7 +75,114 @@ impl GatewayActor {
             }
         });
 
-        Ok(Self { client })
+        let actor = Self {
+            client,
+            lnrpc,
+            task_group,
+        };
+
+        actor.subscribe_htlcs().await?;
+
+        Ok(actor)
+    }
+
+    async fn subscribe_htlcs(&self) -> Result<()> {
+        let actor = self.to_owned();
+        let lnrpc_copy = self.lnrpc.to_owned();
+        let short_channel_id = self.client.config().mint_channel_id;
+        let mut tg = self.task_group.clone();
+
+        let mut stream = lnrpc_copy
+            .subscribe_intercept_htlcs(SubscribeInterceptHtlcsRequest { short_channel_id })
+            .await?;
+
+        tg.spawn(
+            "Subscribe to intercepted HTLCs in stream",
+            move |subscription| async move {
+                while let Some(SubscribeInterceptHtlcsResponse {
+                    payment_hash,
+                    outgoing_amount_msat,
+                    intercepted_htlc_id,
+                    ..
+                }) = match stream.message().await {
+                    Ok(Some(msg)) => Some(msg),
+                    Ok(None) => {
+                        warn!("HTLC stream closed by service");
+                        None
+                    }
+                    Err(e) => {
+                        error!("HTLC stream closed with error: {:?}", e);
+                        None
+                    }
+                } {
+                    if subscription.is_shutting_down() {
```

Comment:
If I understand this correctly, we only check if our task group is shutting down after we asynchronously received a message with `stream.message()`. Hence, as long as we didn't receive a message, we're not checking for shutdown. 

I think we want to do it as in the [`peers.rs`](https://github.com/fedimint/fedimint/blob/master/fedimint-server/src/net/peers.rs#L163-L169). We create a shutdown handler and then use `tokio::select` to choose where to make progress; also note the `while` loop for the shutdown check.
[Another similar example is used in `task.rs`](https://github.com/fedimint/fedimint/blob/master/fedimint-api/src/task.rs#L126-L148)

Reply:
[pending]

Reply:
We could solve this as a follow-up. It's still broken everywhere else too.